src/login/logind-button.c

```diff
@@ -386,17 +475,17 @@ static int button_suitable(int fd) {
                 return -errno;
 
         if (bitset_get(types, EV_KEY)) {
-                unsigned long keys[CONST_MAX5(KEY_POWER, KEY_POWER2, KEY_SLEEP, KEY_SUSPEND, KEY_RESTART)/ULONG_BITS+1];
+                unsigned long keys[KEY_MAX_INTERESTED/ULONG_BITS+1];
 
                 if (ioctl(fd, EVIOCGBIT(EV_KEY, sizeof keys), keys) < 0)
                         return -errno;
 
-                if (bitset_get(keys, KEY_POWER) ||
-                    bitset_get(keys, KEY_POWER2) ||
-                    bitset_get(keys, KEY_SLEEP) ||
-                    bitset_get(keys, KEY_SUSPEND) ||
-                    bitset_get(keys, KEY_RESTART))
-                        return true;
+                FOREACH_ARRAY(key, keys_interested, ELEMENTSOF(keys_interested)) {
+                        assert(*key <= KEY_MAX_INTERESTED);
+
+                        if (bitset_get(keys, *key))
+                                return true;
```

Comment:
hmm, this isn't really right.

this check is suppsoed to filter out kbds that lack the keys we are interested in. but if a kbd has shift/alt/ctrl, but no esc you indicate "yes, we are interested", but that's actually not true.

Reply:
Ah, OK. That makes sense